x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java

```diff
@@ -253,6 +255,16 @@ private void testEvaluate(boolean readFloating) {
         }
     }
 
+    private Object toJavaObjectUnsignedLongAware(Block block, int position) {
```

Comment:
Wondering if we'll want to do this sort of conversion for other (future) types, in which case `toJavaObjectTypeAware()` might be more suitable. But I guess this can be updated then.

Reply:
it's tricky, because the base `toJavaObject` function is already somewhat type aware; it just renders unsigned longs as encoded longs.  I'm open to renaming this, but I agree waiting until we need to add more functionality to it makes more sense.